core-web/libs/portlets/edit-ema/portlet/src/lib/utils/fake-page.ts

```diff
@@ -0,0 +1,1899 @@
+              <form
+                class="rd-form form-sm rd-mailform"
+                data-form-output="form-output-global"
+                data-form-type="contact"
+                method="post"
+                action="">
+                <div class="form-wrap">
+                  <input
+                    class="form-input"
+                    id="footer-form-name"
+                    type="text"
+                    name="name"
+                    data-constraints="@Required" />
+                  <label class="form-label" for="footer-form-name">Name</label>
+                </div>
+                <div class="form-wrap">
+                  <input
+                    class="form-input"
+                    id="footer-form-email"
+                    type="email"
+                    name="email"
+                    data-constraints="@Email @Required" />
+                  <label class="form-label" for="footer-form-email">E-mail</label>
+                </div>
+                <div class="form-wrap">
+                  <label class="form-label" for="footer-form-message">Message</label>
+                  <textarea
+                    class="form-input"
+                    id="footer-form-message"
+                    name="message"
+                    data-constraints="@Required"></textarea>
+                </div>
+                <div class="form-wrap"><button class="button button-md button-primary" type="submit">Send</button></div>
+              </form>
+            </div>
+          </div>
+        </div>
+      </div>
+      <div class="container">
+        <p class="mb-0 rights">
+          <span>©2012 - 2024 All Rights Reserved</span>
+          <a href="https://dotcms.com/company/policies/privacy-policy">Terms of Use</a>
+          <span>and</span>
+          <a href="https://dotcms.com/company/policies/privacy-policy">Privacy Policy</a>
+        </p>
+      </div>
+    </footer>
+    <script src="/application/themes/travel/js/core.min.js"></script>
+    <script src="/application/themes/travel/js/script.js"></script>
+
+    <script>
+      // THIS IS WHERE WE COMMUNICATE OUR RENDERED PAGET WITH THE EDITOR
+      const contentlet = document.querySelector('[data-dot="contentlet"]');
+
+      contentlet.addEventListener('pointerenter', (e) => {
+
+        let target = e.target;
+
+        if (target.dataset.dot !== 'contentlet') {
+            target = target.closest('[data-dot="contentlet"]');
+        }
+
+        if (!target) {
+            return;
+        }
+
+        const { x, y, width, height } = target.getBoundingClientRect();
+
+        const contentletPayload = JSON.parse(target.dataset.content ?? '{}');
+
+        window.parent.postMessage({
+            action: 'set-contentlet',
+            payload: {
+                x,
+                y,
+                width,
+                height,
+                payload: contentletPayload
+            }
+        }, '*');
+      });
```

Comment:
And we inject the JavaScript to emit the events... now the tricky part is the most of this code will be the same as we do in react, we need to find a way to share the code so we don't have to write and test it twice.

Reply:
We can try to make it as reusable as possible and move it to the `client`